js/core/draft.js

```diff
@@ -108,6 +108,108 @@ define(["dao", "globals", "ui", "core/finances", "core/player", "core/team", "li
         });
     }
 
+    function lotteryLogTxt(tid, type, number) {
+        var txt = 'The <a href="' + helpers.leagueUrl(["roster", g.teamAbbrevsCache[tid], g.season]) + '">' + g.teamNamesCache[tid] + '</a>';
+        if (type === 'chance') {
+            txt += " has a " + number + "% chance of getting the top overall pick of the "+ g.season +" draft.";
+        } else if ( type === 'movedup') {
+            txt += " moved up the lottery and will select " + helpers.ordinal(number) + " overall in the " + g.season + " draft.";
+        } else if ( type === 'moveddown') {
+            txt += " moved down the lottery and will select " + helpers.ordinal(number) + " overall in the " + g.season + " draft.";
+        } else if ( type === 'normal') {
+            txt += " will select " + helpers.ordinal(number) + " overall in the " + g.season + " draft.";
+        }
+        return txt;
+    }
+
+    function logAction(tid, txt) {
+        eventLog.add(null, {
+            type: "draft",
+            text: txt,
+            showNotification: tid === g.userTid,
+            pids: [],
+            tids: [tid]
+        });
+    }
+
+    function logLotteryChances(chances, teams, draftOrder) {
+        var i, chance, total, txt, origTm, tm;
+
+        for (i=0; i<chances.length; i++) {
+            origTm = teams[i].tid;
+            tm = draftOrder[origTm][1].tid;
+            txt = lotteryLogTxt(tm, 'chance', helpers.round(chances[i], 2));
+            logAction(tm, txt);
+        }
+    }
+
+    function logLotteryWinners(chances, teams, tm, origTm, pick) {
+        var idx, txt, i;
+        for (i=0; i< teams.length; i++) {
+            if (teams[i].tid === origTm) {
+                idx = i;
+                break;
+            }
+        }
+        if (chances[idx] < chances[pick-1]) {
+            txt = lotteryLogTxt(tm, 'movedup', pick);
+        } else if (chances[idx] > chances[pick-1]) {
+            txt = lotteryLogTxt(tm, 'moveddown', pick);
+        } else {
+            txt = lotteryLogTxt(tm, 'normal', pick);
+        }
+        logAction(tm, txt);
+    }
+
+    /**
+    * Divide the combinations between teams with tied records.
+    *
+    * If isFinal is true, the remainder value is distributed randomly instead
+    * of being set as a decimal value on the result.
+    */
+    function updateChances(chances, teams, isFinal) {
```

Comment:
Unit tests of this would be good, to show it is actually working correctly, including for cases like 3 way ties.